mdata/chunk/archive/archive.go

```diff
@@ -0,0 +1,69 @@
+package archive
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/raintank/metrictank/mdata/chunk"
+	"gopkg.in/raintank/schema.v1"
+)
+
+//go:generate msgp
+type Archive struct {
+	RowKey          string
+	SecondsPerPoint uint32
+	Points          uint32
+	Chunks          []chunk.IterGen
+}
+
+//go:generate msgp
+type Metric struct {
+	MetricData        schema.MetricData
```

Comment:
so in principle we only need a schema.MetricDefinition here.
MetricData is an extension of MetricDefinition which has a time and value attribute. e.g. it holds the definition and can hold one point. Generating the id is equivalent for both types.  we don't need to pass a point here, since all data is in the itergens.  However, the index only exposes an AddOrUpdate method that assumes incoming MetricData types, not MetricDefinition. so I guess it's ok to keep it like this, it's the most pragmatic. just not perfect from a purist POV.